pydantic/type_adapter.py

```diff
@@ -233,100 +208,122 @@ def __init__(
         self._type = type
         self._config = config
         self._parent_depth = _parent_depth
-        if module is None:
-            f = sys._getframe(1)
-            self._module_name = cast(str, f.f_globals.get('__name__', ''))
-        else:
-            self._module_name = module
-
-        self._core_schema: CoreSchema | None = None
-        self._validator: SchemaValidator | PluggableSchemaValidator | None = None
-        self._serializer: SchemaSerializer | None = None
-
-        if not self._defer_build():
-            # Immediately initialize the core schema, validator and serializer
-            with self._with_frame_depth(1):  # +1 frame depth for this __init__
-                # Model itself may be using deferred building. For backward compatibility we don't rebuild model mocks
-                # here as part of __init__ even though TypeAdapter itself is not using deferred building.
-                self._init_core_attrs(rebuild_mocks=False)
-
-    @contextmanager
-    def _with_frame_depth(self, depth: int) -> Iterator[None]:
-        self._parent_depth += depth
-        try:
-            yield
-        finally:
-            self._parent_depth -= depth
 
-    @_frame_depth(1)
-    def _init_core_attrs(self, rebuild_mocks: bool) -> None:
+        self.core_schema: CoreSchema
+        self.validator: SchemaValidator | PluggableSchemaValidator
+        self.serializer: SchemaSerializer
+
+        localns: _namespace_utils.MappingNamespace = (
+            _typing_extra.parent_frame_namespace(parent_depth=self._parent_depth) or {}
+        )
+        globalns: _namespace_utils.MappingNamespace = sys._getframe(max(self._parent_depth - 1, 1)).f_globals
```

Comment:
If we also use the `_parent_depth` attribute to fetch the global ns, we'll need to update the docstring for `_parent_depth` in `__init__`